core/src/main/java/org/apache/calcite/util/Util.java

```diff
@@ -481,22 +481,26 @@ public static void print(
    * prints <code>"x\"y"</code>.
    */
   public static void printJavaString(
-      PrintWriter pw,
+      Appendable pw,
       String s,
       boolean nullMeansNull) {
-    if (s == null) {
-      if (nullMeansNull) {
-        pw.print("null");
+    try {
+      if (s == null) {
+        if (nullMeansNull) {
+          pw.append("null");
+        }
+      } else {
+        String s1 = replace(s, "\\", "\\\\");
+        String s2 = replace(s1, "\"", "\\\"");
+        String s3 = replace(s2, "\n\r", "\\n");
+        String s4 = replace(s3, "\n", "\\n");
+        String s5 = replace(s4, "\r", "\\r");
+        pw.append('"');
+        pw.append(s5);
+        pw.append('"');
       }
-    } else {
-      String s1 = replace(s, "\\", "\\\\");
-      String s2 = replace(s1, "\"", "\\\"");
-      String s3 = replace(s2, "\n\r", "\\n");
-      String s4 = replace(s3, "\n", "\\n");
-      String s5 = replace(s4, "\r", "\\r");
-      pw.print("\"");
-      pw.print(s5);
-      pw.print("\"");
+    } catch (IOException ioe) {
+      throw new RuntimeException("Propagates exception in cause.", ioe);
```

Comment:
Perhaps [UncheckedIOException](https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html) is better in this case ?

Reply:
I didn't know about UncheckedIOException, yes it seems more appropriate. Thanks for the pointer :)